In [63]:
import pandas as pd

__Описание признаков__

`id` -  уникальный номер объявления

`name` -  название объявления

`host_id` -  уникальный номер владельца жилья

`host_name` -  имя владельца жилья

`neighbourhood` -  район Нью-Йорка

`room_type` -  тип жилья, о которой идет речь в объявлении

`price` -  цена в долларах

`minimum_nights` -  минимальное число ночей, на которое сдается комната

`number_of_reviews` -  число отзывов

`reviews_per_month` -  число отзывов в месяц

In [64]:
df = pd.read_csv('../data/nyc_airbnb.csv')
df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood        room_type  price  minimum_nights  \
0         John      Brooklyn     Private room    149               1   
1     Jennifer     Manhattan  Entire home/apt    225               1   
2    Elisabeth     Manhattan     Private room    150               3   
3  LisaRoxanne      Brooklyn  Entire home/apt     89               1   
4        Laura     Manhattan  Entire home/apt     80              10   

   number_of_reviews  reviews_per_month  
0                  9               0.21  
1                 45               0.38  
2                  0                NaN  
3                270               4.64  
4                  9               0.10

In [65]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 48895 entries, 0 to 48894
Data columns (total 10 columns):
id                   48895 non-null int64
name                 48879 non-null object
host_id              48895 non-null int64
host_name            48874 non-null object
neighbourhood        48895 non-null object
room_type            48895 non-null object
price                48895 non-null int64
minimum_nights       48895 non-null int64
number_of_reviews    48895 non-null int64
reviews_per_month    38843 non-null float64
dtypes: float64(1), int64(5), object(4)
memory usage: 3.7+ MB


## Вопросы

1. Какая средняя суточная цена жилья в Нью-Йорке?

In [66]:
# Ваш код здесь

2. В каком районе чаще всего сдают жилье в аренду?

In [67]:
# Ваш код здесь

3. Какая средняя суточная цена жилья для каждого района города?

In [68]:
# Ваш код здесь

4. В данных есть всего три типа сдаваемого жилья:

In [69]:
df['room_type'].unique()

array(['Private room', 'Entire home/apt', 'Shared room'], dtype=object)

`Private room` - комната целиком сдается постояльцу,

`Entire home/apt` - постояльцу сдается целиком дом или квартира,

`Shared room` - постоялец будет делить комнату с другими людьми

Для каждого района Нью-Йорка, ответьте на вопрос, где какой тип жилья чаще всего мелькает в объявлениях?

In [70]:
# Ваш код здесь

5. Какая средняя цена для разных типов комнат в зависимости от района (воспользуйтесь методом `groupby`)?

In [71]:
# Ваш код здесь

6. Какие медианные значения у минимального числа ночей, на которое сдают жилье (признак `minumum_nights`) в зависимости от района?

In [72]:
# Ваш код здесь

7. Какой `host_id` у владельца жилья, который чаще всех давал объявления?

In [73]:
# Ваш код здесь

8. Есть ли жилье со стоимостью больше 150 долларов, у которого больше 400 отзывов?

In [84]:
# Ваш код здесь